google/cloud/storage/testbench/gcs_object.py

```diff
@@ -424,13 +424,14 @@ def patch_revision(self, request):
         patch = json.loads(request.data)
         writeable_keys = {
             'acl', 'cacheControl', 'contentDisposition', 'contentEncoding',
-            'contentLanguage', 'contentType', 'metadata'
+            'contentLanguage', 'contentType', 'eventBasedHold', 'metadata',
+            'temporaryHold'
         }
         for key, value in patch.iteritems():
             if key not in writeable_keys:
                 raise error_response.ErrorResponse(
                     'Invalid metadata change. %s is not writeable' % key,
-                    status_code=503)
+                    status_code=412)
```

Comment:
I don't think 412 applies in most cases, unless I'm reading this wrong.  Is this the error code you get back when you try to remove an object that's under hold?

Reply:
No. I was just trying to create some error if the library sent a non-writable field, but you are right, trying to stay closer to the behavior of the service is probably better.

I changed the code to ignore the non-writable keys, we could also check against the existing values, but that is probably an improvement to the testbench unrelated to this PR?

Reply:
Indeed. Works for me.